### PR TITLE
docs: databases/rest http tutorials

### PR DESCRIPTION
**Pull request by joshua-mo-143**
* Adds the "Databases with rust" and "REST HTTP service with axum" documents.

### CODE REVIEW

**Comment by ivancernja** [comment]
@oddgrd Wanna give it a quick read-through? :)

File: tutorials/rest-http-service-with-axum.mdx
Comment (oddgrd) [inline tutorials/rest-http-service-with-axum.mdx:355]: Nit: Having this here might be a bit confusing? :thinking:

File: tutorials/rest-http-service-with-axum.mdx
Comment (oddgrd) [inline tutorials/rest-http-service-with-axum.mdx:296]: I think it would be more useful if this example checked something on the request, rather than matching on a hardcoded string.

**Review by oddgrd — COMMENTED** [review]
Overall this looks really great and I think it will be useful for a lot of people, thanks @joshua-mo-143! :star_struck:  I left a few nits, which you can take or leave, but the main thing missing is just formatting of code blocks. LGTM when this is addressed!

File: tutorials/rest-http-service-with-axum.mdx
Comment (joshua-mo-143) [reply tutorials/rest-http-service-with-axum.mdx:279]: Hmmmm. We probably should since it'll make it easier to have a point of reference.

File: tutorials/rest-http-service-with-axum.mdx
Comment (joshua-mo-143) [reply tutorials/rest-http-service-with-axum.mdx:355]: Hmmmmm. Yeah, probably. I'll amend this

Comment (joshua-mo-143) [reply tutorials/rest-http-service-with-axum.mdx:355]: Have been amended, once I push a new commit this should be fixed

File: tutorials/rest-http-service-with-axum.mdx
Comment (joshua-mo-143) [reply tutorials/rest-http-service-with-axum.mdx:279]: Has been amended, I've added a link to Axum middleware at the bottom of the section which you'll see in the next commit

File: tutorials/rest-http-service-with-axum.mdx
Comment (joshua-mo-143) [reply tutorials/rest-http-service-with-axum.mdx:251]: Has been amended for next commit, I didn't realise that it was a re-import.

File: tutorials/databases-with-rust.mdx
Comment (joshua-mo-143) [reply tutorials/databases-with-rust.mdx:1]: No worries, will get this done - originally I brought it over straight from rust fmt to my blog article and then it needed re-formatting as the code looked a bit derpy, so I knew there was probably something that needed to be done at some point with it.

File: tutorials/rest-http-service-with-axum.mdx
Comment (joshua-mo-143) [reply tutorials/rest-http-service-with-axum.mdx:31]: It looks like the rest of the docs use non-capitalised sentences. I wouldn't mind amending this but it would probably require maybe changing the rest of the docs for consistency

File: tutorials/databases-with-rust.mdx
Comment (joshua-mo-143) [reply tutorials/databases-with-rust.mdx:59]: Amended, see next commit

File: tutorials/databases-with-rust.mdx
Comment (joshua-mo-143) [reply tutorials/databases-with-rust.mdx:76]: Amended, see next commit

File: tutorials/databases-with-rust.mdx
Comment (joshua-mo-143) [reply tutorials/databases-with-rust.mdx:1]: Amended, see next commit

File: tutorials/databases-with-rust.mdx
Comment (joshua-mo-143) [reply tutorials/databases-with-rust.mdx:127]: Amended, see next commit

File: tutorials/rest-http-service-with-axum.mdx
Comment (joshua-mo-143) [reply tutorials/rest-http-service-with-axum.mdx:1]: Amended, see next commit

File: tutorials/databases-with-rust.mdx
Comment (joshua-mo-143) [reply tutorials/databases-with-rust.mdx:86]: This makes more sense, will amend.

File: tutorials/rest-http-service-with-axum.mdx
Comment (joshua-mo-143) [reply tutorials/rest-http-service-with-axum.mdx:296]: Ok no worries, I'll amend this

**Comment by joshua-mo-143** [comment]
Have made some amendments, should be OK to go I think once these are approved

File: tutorials/rest-http-service-with-axum.mdx
Comment (oddgrd) [reply tutorials/rest-http-service-with-axum.mdx:31]: No worries!

File: tutorials/databases-with-rust.mdx
Comment (oddgrd) [inline tutorials/databases-with-rust.mdx:56]:
```suggestion
let query = sqlx::query("SELECT * FROM users WHERE id = $1")
```

File: tutorials/databases-with-rust.mdx
Comment (oddgrd) [inline tutorials/databases-with-rust.mdx:195]:
```suggestion
    let mut data = state.persist.load::<Data>("data");
    
    data.people.push( Person { name: req.name, age: req.age } );
    data.total_records += 1;
    
    match state.persist.save::<Data>("data", data) {
        Ok(_) => Ok((StatusCode::OK, "Successful!".to_string())),
        Err(e) => Err((StatusCode::INTERNAL_SERVER_ERROR, e.to_string()))
    }
```

File: tutorials/rest-http-service-with-axum.mdx
Comment (oddgrd) [inline tutorials/rest-http-service-with-axum.mdx:27]: This needs indentation, same for a few blocks below this

File: tutorials/rest-http-service-with-axum.mdx
Comment (oddgrd) [inline tutorials/rest-http-service-with-axum.mdx:135]:
```suggestion
        .route("/", get(hello_world)).with_state(state)
```

File: tutorials/rest-http-service-with-axum.mdx
Comment (oddgrd) [inline tutorials/rest-http-service-with-axum.mdx:194]: This needs formatting too (weird indentation)

File: tutorials/rest-http-service-with-axum.mdx
Comment (oddgrd) [inline tutorials/rest-http-service-with-axum.mdx:241]:
```suggestion
```

File: tutorials/rest-http-service-with-axum.mdx
Comment (oddgrd) [inline tutorials/rest-http-service-with-axum.mdx:105]:
```suggestion
```

File: tutorials/databases-with-rust.mdx
Comment (oddgrd) [inline tutorials/databases-with-rust.mdx:186]: The first line of functions shouldn't be a blank line
```suggestion
```

**Review by oddgrd — COMMENTED** [review]
Hey Josh, just some more formatting nits and we're good to go. I know this can be a pain in markdown, so thanks for fixing this up!

**Comment by joshua-mo-143** [comment]
I've made the required amendments.

I also fixed a code snippet that used the `tokio::main` macro instead of `shuttle_runtime::main`.

**Review by oddgrd — APPROVED** [review]
Thanks!

**Pull request merged**
